internal/scanner/walk.go

```diff
@@ -117,6 +117,18 @@ func (w *Walker) walkAndHashFiles(fchan chan protocol.FileInfo) filepath.WalkFun
 		// Return value used when we are returning early and don't want to
 		// process the item. For directories, this means do-not-descend.
 		var skip error // nil
+
```

Comment:
Ah. This section needs to be added after the error checks on (current) line 137-142, or `info` may be nil and this explodes.